owl/SOMA-IO.owl

```diff
@@ -25,40 +124,321 @@
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#isDefinedIn"/>
+                <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Client-Server_Specification"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:label xml:lang="en">Client role</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Computer_Code -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Computer_Code">
+        <owl:equivalentClass>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Structured_Text"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#isGivenMeaningBy"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Programming_Language"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </owl:equivalentClass>
+        <owl:equivalentClass>
+            <owl:Class>
+                <owl:intersectionOf rdf:parseType="Collection">
+                    <rdf:Description rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Structured_Text"/>
+                    <owl:Restriction>
+                        <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#expresses"/>
+                        <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Algorithm"/>
+                    </owl:Restriction>
+                </owl:intersectionOf>
+            </owl:Class>
+        </owl:equivalentClass>
+        <rdfs:comment xml:lang="en">The Program itself (the specific set of instruction in a Programming Language), not the file that it is contained in nor the implemented algorithm!</rdfs:comment>
+        <rdfs:label xml:lang="en">Computer Program</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Computer_Language -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Computer_Language">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#FormalLanguage"/>
+        <rdfs:comment xml:lang="en">A computer language is a formal language used in communication with a computer.
+
+From Wikipedia: https://en.wikipedia.org/wiki/Computer_language</rdfs:comment>
+        <rdfs:label xml:lang="en">Computer Language</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Database -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Database">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#SoftwareRole"/>
```

Comment:
Is this really intended as SoftwareRole and not Software?

Reply:
Yes! Essentially, you could use an SQL Database also as a clock without ever storing any data: Just send a query and you will get the time. However, to most, it will have the Role of a Database.